src/HIE/Bios/Environment.hs

```diff
@@ -223,7 +225,42 @@ disableOptimisation df = updOptLevel 0 df
 getTargetsFromGhciScript :: FilePath -> IO [String]
 getTargetsFromGhciScript script = do
   contents <- lines <$> readFile script
-  return
-    $ concatMap (tail {- first element is ":add" which we want to remove -}
-                      . words)
-    $ filter (":add" `isPrefixOf`) contents
+  let parseGhciLine = concatMap fst . filter (null . snd) . readP_to_S parser
```

Comment:
Should it perhaps be an error if the parse fails?

Reply:
I dont think so, since an example ghci-script from stack looks like:
```
:add A B
:module + B
```
We only want to successfully parse the first line. 
However, we could fail only for lines beginning with `:add`.